dds/DCPS/WriteDataContainer.h

```diff
@@ -528,6 +531,28 @@ class OpenDDS_Dcps_Export WriteDataContainer : public PoolAllocationBase {
   DDS::DurabilityServiceQosPolicy const & durability_service_;
 
 #endif
+
+  /// Timer responsible for reporting missed offered deadlines.
+  RcHandle<DCPS::PmfSporadicTask<WriteDataContainer> > deadline_task_;
+  TimeDuration deadline_period_; // TimeDuration::zero_value means no deadline.
+  typedef std::multimap<MonotonicTimePoint, PublicationInstance_rch> DeadlineMapType;
+  DeadlineMapType deadline_map_;
+
+  /// Lock for synchronization of @c status_ member.
+  ACE_Recursive_Thread_Mutex& deadline_status_lock_;
+  /// Reverse lock used for releasing the @c status_lock_ listener upcall.
+  ACE_Reverse_Lock<ACE_Recursive_Thread_Mutex> deadline_reverse_status_lock_;
```

Comment:
Do we need this as a member? Can we just create these inline as we need them?

Reply:
I'll inline it.